internal/lm/nvml.go

```diff
@@ -218,6 +230,96 @@ func newGPUModeLabeler(devices []resource.Device) (Labeler, error) {
 	return labels, nil
 }
 
+func newImexDomainLabeler(configFile string, device []resource.Device) (Labeler, error) {
+	if configFile == "" {
+		return nil, nil
+	}
+
+	// Read file and parse it
+	_, err := os.Stat(configFile)
+	if os.IsNotExist(err) {
+		return nil, nil
+	}
+	imexConfig, err := os.Open(configFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open imex config file: %v", err)
+	}
+	defer imexConfig.Close()
+
+	// check if the file is empty
+	stat, err := imexConfig.Stat()
+	if err != nil {
+		return nil, nil
+	}
+	if stat.Size() == 0 {
+		return nil, nil
+	}
+
+	// Read the file line by line
+	var ips []string
+	scanner := bufio.NewScanner(imexConfig)
+	for scanner.Scan() {
+		ip := strings.TrimSpace(scanner.Text())
+		if net.ParseIP(ip) == nil {
+			return nil, fmt.Errorf("invalid IP address in imex config file: %s", ip)
+		}
+		ips = append(ips, ip)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("failed to read imex config file: %v", err)
+	}
+
+	// Sort the IP addresses
+	sort.Strings(ips)
+
+	// Join the sorted IPs into a single string
+	sortedIPs := strings.Join(ips, "\n")
+
+	hashedconfig := generateUUIDs(sortedIPs)
+
+	var commonClusterUUID string
+	var commonCliqueID string
+	for _, d := range device {
+		// Skip non NVML devices
```

Comment:
drop comment -- it may be the case that other devices become IMEX capable in the future and the code itself is clear enough here.